source/features/collapse-markdown-sections.ts

```diff
@@ -0,0 +1,46 @@
+import select from 'select-dom';
+import delegate from 'delegate-it';
+import * as pageDetect from 'github-url-detection';
+
+import features from '.';
+
+function toggleSection(sectionHeading: HTMLElement): void {
+	const sectionState = sectionHeading.classList.toggle('rgh-markdown-section-collapsed');
+	let element = sectionHeading.nextElementSibling as HTMLElement;
```

Comment:
If we use this, we don't need `toggleAllTopSections` and the secondary delegate

```suggestion
	if (sectionHeading.tagName === 'H1') {
		  for (const element of sectionHeading.parentElement!.children) {
	 		  if (/^H\d$/.test(element.tagName)) {
	 			  element.classList.toggle('rgh-markdown-section-collapsed', sectionState);
	 		  } else {
	 			  element.hidden = sectionState;
	 		  }
	 	  }
	 	  return;
	}

	let element = sectionHeading.nextElementSibling as HTMLElement;
```

Reply:
Better yet, if you set `let element` to pick up the `.firstElementChild` you could use the existing loop, except that you'd also need to skip the `<=` comparison there.

Reply:
> if change the element you could use the existing loop

Yeah I though about that but I wasn't sure making the code messier was warranted by the performance gains (since there's _usually_ just one or two `h1` per Markdown document). But it does makes the feature quite shorter.